sympy/matrices/matrices.py

```diff
@@ -2402,21 +2435,21 @@ def dot(self, b):
                     type(b))
 
         mat = self
-        if mat.cols == b.rows:
-            if b.cols != 1:
-                mat = mat.T
-                b = b.T
-            prod = flatten((mat * b).tolist())
-            if len(prod) == 1:
-                return prod[0]
-            return prod
-        if mat.cols == b.cols:
-            return mat.dot(b.T)
-        elif mat.rows == b.rows:
-            return mat.T.dot(b)
-        else:
-            raise ShapeError("Dimensions incorrect for dot product: %s, %s" % (
-                self.shape, b.shape))
+        if (1 not in mat.shape) or (1 not in b.shape) :
+            SymPyDeprecationWarning(
+                feature="dot() no longer support dot of (m,n) and (1,n) matrices.",
+                issue=13815,
+                deprecated_since_version="1.2").warn()
```

Comment:
The warning looks like: `dot() no longer support dot of (m,n) and (1,n) matrices. has been
deprecated since SymPy 1.2. See
https://github.com/sympy/sympy/issues/13815 for more info.`. The `feature` should be a noun.